### PR TITLE
Add giscus to blog posts

### LINKED ISSUE #1208: Support blog comments
It would be nice if one could add comments on each blog post for feedback. See https://giscus.app/

### PR DESCRIPTION
**Pull request by gastaldi**
Fixes #1208

DON'T MERGE THIS YET, JUST TESTING

Discussions created in https://github.com/gastaldi/sample-qualifier/discussions/categories/announcements

### CODE REVIEW

**Comment by maxandersen** [comment]
requires [quarkusio/](https://github.com/quarkusio/quarkus/pull/19479)

**Comment by maxandersen** [comment]
this is definitely all win imo and i would enable discussions just to get this :) 

I wonder if we should run our own giscuss instance but I would say lets start with giscuss.app as users just need to agree once and if they don't want to can go to the github discussion itself.

File: _layouts/post.html [hunk 1]
```diff
@@ -42,3 +42,16 @@ <h1 class="post-title">{{ page.title }}</h1>
...
 </div>
+<script src="https://giscus.app/client.js"
+        data-repo="gastaldi/sample-qualifier"
+        data-repo-id="MDEwOlJlcG9zaXRvcnk0MDAyMjMxMTY="
+        data-category="Announcements"
+        data-category-id="DIC_kwDOF9rrjM4B_2uZ"
+        data-mapping="og:title"
+        data-reactions-enabled="1"
+        data-emit-metadata="0"
+        data-theme="light"
```
Comment (maxandersen) [inline _layouts/post.html:53]: we should probably use one of the dark modes to match closer to the site css.

**Comment by gastaldi** [comment]
> lets start with giscuss.app as users just need to agree once and if they don't want to can go to the github discussion itself.

👍🏻

**Comment by gsmet** [comment]
I had a look at this. I think it's a nice addition and has a lot of value but I have a couple of comments:

* is it such a good idea to use a dark theme, given posts are on a light background? It's quite weird and a few things are not working very well (such as `3 comments` at the top here: https://quarkusio-quarkusio-github-io-preview-pr-1209.surge.sh/blog/quarkus-2-4-2-final-released/ ).
* we need some additional margin/padding at the bottom between the last comment and the footer

**Comment by gastaldi** [comment]
> * is it such a good idea to use a dark theme, given posts are on a light background? It's quite weird and a few things are not working very well (such as `3 comments` at the top here: [quarkusio-quarkusio-github-io-preview-pr-1209.surge.sh/blog/quarkus-2-4-2-final-released](https://quarkusio-quarkusio-github-io-preview-pr-1209.surge.sh/blog/quarkus-2-4-2-final-released/) ).

I agree that the light theme suits better in this case, I'll revert the change

**Comment by gastaldi** [comment]
> * we need some additional margin/padding at the bottom between the last comment and the footer

@gsmet Done

**Comment by gsmet** [comment]
Is it really fixed? Maybe I have a cache issue but I still don't see a space between the end of the comment block and the footer.
Screenshot 2021-11-25 at 17-49-06 Quarkus 2 4 2 Final released - Maintenance release

**Comment by gastaldi** [comment]
@gsmet it is being shown in your image. It has 10px only, should it be higher?

**Comment by gastaldi** [comment]
Perfect! 👌🏻

**Comment by gsmet** [comment]
Cool :)

**Comment by maxandersen** [comment]
i've merged this into #1242 so it goes out together.

**Pull request closed without merging**